### PR TITLE
Add the cluster-api-provider-openstack image

### PR DESCRIPTION
**Pull request by flaper87**
We've forked `cluster-api-provider-openstack` under the `openshift` org to be able to generate image builds and cut them from a release branch. This patch should add the required logic to generate and build this image. 

I'll be adding the respective patch to https://github.com/openshift/ci-operator-prowgen soon

### CODE REVIEW

File: ci-operator/config/openshift/cluster-api-provider-openstack/openshift-cluster-api-provider-openstack-master.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  tag_overrides: {}
+build_root:
+  image_stream_tag:
+    cluster: https://api.ci.openshift.org
+    name: release
+    namespace: openshift
+    tag: golang-1.10
+tests:
+- as: unit
+  commands: make test
```
Comment (stevekuznetsov) [inline ci-operator/config/openshift/cluster-api-provider-openstack/openshift-cluster-api-provider-openstack-master.yaml]: This is the odd man out, does it need `IS_CONTAINER=TRUE`?

**Comment by stevekuznetsov** [comment]
> I'll be adding the respective patch to https://github.com/openshift/ci-operator-prowgen soon

What patch is necessary?

**Comment by flaper87** [comment]
> What patch is necessary?

Honestly, I still don't know. I was going to dive into it now but I was told I'd have to modify that project too. I'm quite new to this org so still figuring some things out.

File: ci-operator/config/openshift/cluster-api-provider-openstack/openshift-cluster-api-provider-openstack-master.yaml (hunk 1 above)
Comment (flaper87) [reply ci-operator/config/openshift/cluster-api-provider-openstack/openshift-cluster-api-provider-openstack-master.yaml]: I copied this file from the AWS one. What is `IS_CONTAINER` used for?

**Comment by flaper87** [comment]
@stevekuznetsov thanks a lot for the review. I'll check what changes are needed to `plugins.yaml`

**Comment by flaper87** [comment]
@stevekuznetsov I've modified `plugins.yaml`

File: ci-operator/config/openshift/cluster-api-provider-openstack/openshift-cluster-api-provider-openstack-master.yaml (hunk 1 above)
Comment (stevekuznetsov) [reply ci-operator/config/openshift/cluster-api-provider-openstack/openshift-cluster-api-provider-openstack-master.yaml]: This field defines what commands get run for the 'unit' test as per the [doc](https://github.com/openshift/ci-operator/blob/master/CONFIGURATION.md#testscommands)

Comment (flaper87) [reply ci-operator/config/openshift/cluster-api-provider-openstack/openshift-cluster-api-provider-openstack-master.yaml]: Thanks for the link to the doc. These commands should be correct for this image too. I'll go through the doc and update the PR accordingly. Thanks again :cake:

**Pull request merged**